rerun_py/rerun_sdk/rerun/log/image.py

```diff
@@ -177,7 +177,9 @@ def log_segmentation_image(
     if not bindings.is_enabled():
         return
 
-    image = np.array(image, dtype=np.uint16, copy=False)
+    image = np.array(image, copy=False)
+    if image.dtype not in (np.uint8, np.uint16):
```

Comment:
I noticed this wasn't hitting the u8 path when I was testing with the minimal segmantation image demo in api_main because it was implicitly converting to uint16.

Reply:
Good catch - let's make a separate PR out of this, since I think we should avoid depending on patched crates and therefore should make the rest of this PR a draft until we have a new `arrow2_convert` release